File: CourseTech.Domain/Dto/Lesson/Practice/LessonPracticeDto.cs
using CourseTech.Domain.Enum;
using CourseTech.Domain.Interfaces.Dtos.Question;

namespace CourseTech.Domain.Dto.Lesson.Practice;

/// <summary>
/// Модель данных для отображения практической части урока пользователю.
/// </summary>
public class LessonPracticeDto
{
    public int LessonId { get; set; }

    public LessonTypes LessonType { get; set; }

    public List<QuestionDtoBase> Questions { get; set; }
}

File: coursetech.client/src/types/dto/question/lesson-practice-dto.ts
import { LessonTypes } from "../../../enums/lesson-types";
import { IQuestionDto } from "./question-dto-interface";

export interface LessonPracticeDto {
    lessonId: number;
    lessonType: LessonTypes;
    questions: IQuestionDto[];
    isDemoMode: boolean;
}